# HG changeset patch
# User George Dunlap <george.dunlap@eu.citrix.com>
# Date 1331197832 0
# Node ID 12c795ed5bec1f5f1c288b5326b9ab991b4f333c
# Parent  56ca2b874928af0352e445ee31aefe5d7fda2f49
hvm: Add command line option to enable ASID support -- on by default

Signed-off-by: George Dunlap <george.dunlap@eu.citrix.com>
Signed-off-by: Keir Fraser <keir@xen.org>

diff --git a/xen/arch/x86/hvm/asid.c b/xen/arch/x86/hvm/asid.c
--- a/xen/arch/x86/hvm/asid.c
+++ b/xen/arch/x86/hvm/asid.c
@@ -25,6 +25,10 @@
 #include <xen/percpu.h>
 #include <asm/hvm/asid.h>
 
+/* Xen command-line option to enable ASIDs */
+static int opt_asid_enabled = 1;
+boolean_param("asid", opt_asid_enabled);
+
 /*
  * ASIDs partition the physical TLB.  In the current implementation ASIDs are
  * introduced to reduce the number of TLB flushes.  Each time the guest's
@@ -62,7 +66,7 @@ void hvm_asid_init(int nasids)
     struct hvm_asid_data *data = &this_cpu(hvm_asid_data);
 
     data->max_asid = nasids - 1;
-    data->disabled = (nasids <= 1);
+    data->disabled = !opt_asid_enabled || (nasids <= 1);
 
     if ( g_disabled != data->disabled )
     {
